crypto/rand/randfile.c

```diff
@@ -101,8 +100,13 @@ int RAND_load_file(const char *file, long bytes)
         else
             n = RAND_FILE_SIZE;
         i = fread(buf, 1, n, in);
+#ifdef EINTR
+        if (i <= 0 && errno != EINTR)
```

Comment:
as far as I remember the discussion on the previous PR, EINTR sets the FILE
into an error state, that must be cleared explicitly, all in all that is possible but....

Reply:
Added a commit to do that, which also fixes a logic error.